specs/auth_SUITE.erl

```diff
@@ -0,0 +1,220 @@
+        }
+      ]
+    }
+  ].
+
+mpx_auth_token_success(Config) ->
+  ConnPid = gun_open(Config),
+  Token = request_token(ConnPid, Config),
+  gun:ws_upgrade(ConnPid, <<"/ws/mpx?token=", Token/binary>>),
+  {gun_ws_upgrade, ConnPid, ok, _Headers} = message(),
+  gun:ws_send(ConnPid, {text, <<"default,sub">>}),
+  {gun_ws, ConnPid, {text, <<"default,open">>}} = message(),
+  ok.
+
+mpx_auth_token_failed(Config) ->
+  ConnPid = gun_open(Config),
+  Token = request_token(ConnPid),
+  gun:ws_upgrade(ConnPid, <<"/ws/mpx?token=", Token/binary>>),
+  {gun_ws_upgrade, ConnPid, ok, _Headers} = message(),
+  gun:ws_send(ConnPid, {text, <<"default,sub">>}),
+  {gun_ws, ConnPid, close} = message(),
+  ok.
+
+default_auth_token_success(Config) ->
+  ConnPid = gun_open(Config),
+  Token = request_token(ConnPid, Config),
+  gun:ws_upgrade(ConnPid, "/ws/mpx"),
+  {gun_ws_upgrade, ConnPid, ok, _Headers} = message(),
+  gun:ws_send(ConnPid, {text, <<"default,sub?token=", Token/binary>>}),
+  {gun_ws, ConnPid, {text, <<"default,open">>}} = message(),
+  ok.
+
+default_auth_token_failed(Config) ->
+  ConnPid = gun_open(Config),
+  Token = request_token(ConnPid),
+  gun:ws_upgrade(ConnPid, "/ws/mpx"),
+  {gun_ws_upgrade, ConnPid, ok, _Headers} = message(),
+  gun:ws_send(ConnPid, {text, <<"default,sub?token=", Token/binary>>}),
+  {gun_ws, ConnPid, close} = message(),
+  ok.
+
+mpx_auth_referer_success(Config) ->
+  ConnPid = gun_open(Config),
+  gun:ws_upgrade(ConnPid, <<"/ws/mpx">>),
+  {gun_ws_upgrade, ConnPid, ok, _Headers} = message(),
+  gun:ws_send(ConnPid, {text, <<"default,sub">>}),
+  {gun_ws, ConnPid, {text, <<"default,open">>}} = message(),
+  ok.
+
+mpx_auth_referer_failed(Config) ->
+  ConnPid = gun_open(Config),
+  gun:ws_upgrade(ConnPid, <<"/ws/mpx">>),
+  {gun_ws_upgrade, ConnPid, ok, _Headers} = message(),
+  {gun_ws, ConnPid, close} = message(),
+  ok.
+
+default_auth_referer_success(Config) ->
+  ConnPid = gun_open(Config),
+  gun:ws_upgrade(ConnPid, "/ws/mpx"),
+  {gun_ws_upgrade, ConnPid, ok, _Headers} = message(),
+  gun:ws_send(ConnPid, {text, <<"default,sub">>}),
+  {gun_ws, ConnPid, {text, <<"default,open">>}} = message(),
+  ok.
+
+default_auth_referer_failed(Config) ->
+  ConnPid = gun_open(Config),
+  gun:ws_upgrade(ConnPid, "/ws/mpx"),
+  #{auth := Any} = ulitos_app:get_var(?APP, default),
+  [{de_auth_referer, ["example.com"]}] = Any,
+  {gun_ws_upgrade, ConnPid, ok, _Headers} = message(),
+  gun:ws_send(ConnPid, {text, <<"default,sub">>}),
+  {gun_ws, ConnPid, close} = message(),
+  ok.
+
+gun_open(Config) ->
+  {ok, ConnPid} = gun:open("localhost", ?config(port, Config)),
+  {gun_up, ConnPid, http} = message(),
+  ConnPid.
+
+request_token(Pid) ->
+  request_token(Pid, <<"">>, <<"">>).
+
+request_token(Pid, Config) ->
+  request_token(Pid, ?config(user, Config), ?config(pass, Config)).
+
+request_token(Pid, User, Pass) ->
+  Headers = [{"Authorization", "Basic " ++ base64:encode_to_string(<<User/binary, ":", Pass/binary>>)}],
```

Comment:
Line 201 is too long:   Headers = [{"Authorization", "Basic " ++ base64:encode_to_string(<<User/binary, ":", Pass/binary>>)}],.